editor/src/clj/editor/app_view.clj

```diff
@@ -2536,116 +2527,33 @@ If you do not specifically require different script states, consider changing th
-                           render-build-error! bob/bundle-bob-commands bob-args project changes-view
-                           (fn [successful?]
-                             (if successful?
-                               (if (some-> output-directory .isDirectory)
-                                 (do
-                                   (when (prefs/get prefs [:bundle :open-output-directory])
-                                     (ui/open-file output-directory))
-                                   (cond
-                                     (and (= :android platform)
-                                          (:adb-install bundle-options)
-                                          (string/includes? (:bundle-format bundle-options) "apk"))
-                                     ;; will eventually close the output
-                                     (adb-post-bundle! prefs output-directory project out (:adb-launch bundle-options))
-
-                                     (and (= :ios platform)
-                                          (:ios-deploy-install bundle-options))
-                                     ;; will eventually close the output
-                                     (ios-post-bundle! prefs output-directory project out (:ios-deploy-launch bundle-options))
-
-                                     :else
-                                     (.close out)))
-                                 (.close out))
-                               (do
-                                 (.close out)
-                                 (dialogs/make-info-dialog
-                                   {:title "Bundle Failed"
-                                    :icon :icon/triangle-error
-                                    :size :large
-                                    :header "Failed to bundle project, please fix build errors and try again"})))))))
-
 (handler/defhandler :bundle :global
-  (run [user-data workspace project prefs app-view changes-view build-errors-view main-stage tool-tab-pane]
-    (let [owner-window (g/node-value app-view :stage)
-          platform (:platform user-data)
-          bundle! (partial bundle! main-stage tool-tab-pane changes-view build-errors-view project prefs platform)]
-      (bundle-dialog/show-bundle-dialog! workspace platform prefs owner-window bundle!))))
+  (options [user-data _context]
+    (when-not user-data
+      (let [contexts [_context]]
+        (->> (handler/realize-menu :editor.bundle/menu)
+             (e/keep
+               (fn [{:keys [command]}]
+                 (when command
+                   (when-let [handler+context (handler/active command contexts true)]
+                     {:command :bundle
+                      :label (handler/label handler+context)
+                      :user-data {:command command
+                                  :handler+context handler+context}}))))
+             (into [])))))
```

Comment:
Use transducer with `into` instead of creating an eduction?

Or you could make use of the new `coll/transfer` function if you prefer the added clarity.